rules/linux/persistence_message_of_the_day_creation.toml

```diff
@@ -0,0 +1,80 @@
+located in the "/etc/update-motd.d/" and "/usr/lib/update-notifier/" directories. These scripts run 
+as the root user every time a user connects over SSH or a serial connection. Adversaries may create
+malicious MOTD files that grant them persistence onto the target every time a user connects to the 
+system by executing a backdoor script or command. This rule detects the execution of potentially 
+malicious processes through the MOTD utility. 
+"""
+from = "now-9m"
+index = ["logs-endpoint.events.*", "auditbeat-*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Potential Persistence Through MOTD File Creation Detected"
+note = """## Triage and analysis
+### Investigating message of the day file creation
+Detection alerts from this rule indicate the creation of a new file in the "/etc/update-motd.d/" or "/usr/lib/update-notifier/ directories. Administrators might alter or create new MOTD files in these directories, so investigation to see whether the file is malicious is vital. The first files to check can be found here:
+- /etc/update-motd.d/
+- /usr/lib/update-notifier/
+
+The file(s) created within these directories may contain links to executables, scripts or may just run a specific command.
+
+To check for the execution of the newly added motd file, the "Suspicious Process Spawned from MOTD Detected" can be checked.
+
+### Response and remediation
+- Initiate the incident response process based on the outcome of the triage.
+- Isolate the involved host to prevent further post-compromise behavior.
+- Investigate credential exposure on systems compromised or used by the attacker to ensure all compromised accounts are identified. Reset passwords for these accounts and other potentially compromised credentials, such as email, business systems, and web services.
+- Delete the MOTD files or restore them to the original configuration.
+- Run a full antimalware scan. This may reveal additional artifacts left in the system, persistence mechanisms, and malware components.
+- Determine the initial vector abused by the attacker and take action to prevent reinfection through the same vector.
+- Leverage the incident response data and logging to improve the mean time to detect (MTTD) and the mean time to respond (MTTR).
+"""
+references = [
+    "https://pberba.github.io/security/2022/02/06/linux-threat-hunting-for-persistence-initialization-scripts-and-shell-configuration/#8-boot-or-logon-initialization-scripts-rc-scripts"
+]
+risk_score = 21
+rule_id = "96d11d31-9a79-480f-8401-da28b194608f"
+severity = "low"
+tags = ["Elastic", "Host", "Linux", "Threat Detection", "Persistence", "Investigation Guide"]
+type = "eql"
+
+query = '''
+sequence by user.id, host.id with maxspan = 15s
+[file where event.type == "creation" and
+file.path : (
+  "/etc/update-motd.d/*",
+  "/usr/lib/update-notifier/*"
+)]
+
+[process where event.type == "start" and 
+event.action == "exec" and
+process.name == "chmod" and
+process.args : "+x"]
```

Comment:
```suggestion
sequence by user.id, host.id with maxspan = 15s
[file where event.type == "creation" and
file.path : (
  "/etc/update-motd.d/*",
  "/usr/lib/update-notifier/*"
)]
[process where event.type == "start" and 
event.action == "exec" and
process.name == "chmod" and
process.args : "+x"
process.args : ("/etc/update-motd.d/*", "/usr/lib/update-notifier/*")]
```

Reply:
Could we add the MOTD file path to the chmod process args to make this more specific and limit FPs?

Reply:
@DefSecSentinel the reason I chose not to do this, is because this will be evaded by moving into the directory and doing a `chmod +x file` without specifying the full path, and in the chmod event there is no file path or other value we can use that will always be available to capture it. WDYT?

Reply:
I agree with Colson, here this can be a bit noisy. The goal is to check if the files created under the motd are made executable, then we could possibly construct args to include those parameters. 
To tackle chmod +x on where there is no file path this could be exclusive OR condition in the args parameter.
This will be noisy

Reply:
@DefSecSentinel @shashank-elastic I changed the query to a normal EQL rule instead of a sequence, by using "inclusion by exception" logic. This logic has triggered 0 false positives in my several linux environments over the last 90 days. Only system processes automate the process of creating these files in these directories. This solves the FP issue present above, in addition to the easy bypass by making a file executable before moving it to the directory or using a different chmod command than +x to make a file executable. Can I get a recheck?